src/api/dependency-discovery/src/dependency-list.js

```diff
@@ -1,14 +1,52 @@
 const { readFile } = require('fs/promises');
 const { join } = require('path');
 const { cwd } = require('process');
+const { getPackument } = require('query-registry');
+
+const dependenciesGlobal = {};
 
 async function getDependencies() {
-  const dependencies = await readFile(join(cwd(), 'deps.txt'), 'utf8');
+  if (Object.keys(dependenciesGlobal).length === 0) {
+    const dependencyList = await readFile(join(cwd(), 'deps.txt'), 'utf8');
+
+    // The order of the alternatives is important!
+    // The regex engine will favor the first pattern on an alternation
+    // even if the other alternatives are subpatterns
```

Comment:
Using `forEach` is better here, https://eslint.org/docs/rules/no-restricted-syntax
> iterators/generators require regenerator-runtime, which is too heavyweight for this guide to allow them. Separately, loops should be avoided in favor of array iterations

For some reason, eslint fails to detect this. I'd say eslint is not working at all, I tried adding an unused variable and eslint tests still pass.

Reply:
cc @cindyledev